.github/workflows/livetest.yml

```diff
@@ -0,0 +1,149 @@
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            github.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: 'Download of the dists artifact failed: Retry again in 10 minutes'
+            })
+        if: steps.make_dist.outcome == 'failure'
+
+      - run: exit 1
+        if: steps.make_dist.outcome == 'failure'
+
+      - name: Install dependencies
+        run: |
+          sudo sed -Ei 's/^# deb-src /deb-src /' /etc/apt/sources.list
+          sudo apt update
+          sudo apt-get -y build-dep cockpit
+          ./tools/make-debs --quick
+          # FIXME: pcp is just utterly slow to start makes the installation fail
+          rm -f cockpit-pcp*.deb
+          for deb in cockpit*.deb; do sudo apt -y install ./$deb; done
+          sudo systemctl start cockpit.socket
+
+      - name: Get the commit SHA that triggered the workflow
+        id: get_commit_sha
+        run: |
+          echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+
+      - name: Get ssh gateway port to forward to
+        id: remote_port
+        env:
+          SHA_SHORT: "${{ steps.get_commit_sha.outputs.sha_short }}"
+        run: |
+            echo ::set-output name=SSH_GATEWAY_PORT::$((16#$SHA_SHORT % 32768 + 1500))
+
+      - name: Write a comment on the PR with link to the cockpit instance
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            github.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: 'Cockpit preview: https://${{ env.RELAY_HOST }}:${{ steps.remote_port.outputs.SSH_GATEWAY_PORT }}'
+            })
+
+      - name: Create an SSH private key from the secret
+        run: |
+          sudo touch /run/SINK_SSHRELAY_PRIVATE && sudo chown $(id -u):$(id -g) /run/SINK_SSHRELAY_PRIVATE
+          echo "${{ secrets.SINK_SSHRELAY_PRIVATE }}" > /run/SINK_SSHRELAY_PRIVATE
+          chmod 600 /run/SINK_SSHRELAY_PRIVATE
+          mkdir -p ~/.ssh &&  ssh-keyscan -H ${{ env.RELAY_HOST }} > ~/.ssh/known_hosts
+
+      - name: Keep ssh connection alive
+        run: sudo sed -i -r 's/#ClientAliveInterval.*/ClientAliveInterval 600/' /etc/ssh/sshd_config
+
+      - name: Forward cockpit port 9090 to a server we can access
+        run: ssh -f -R ':${{ steps.remote_port.outputs.SSH_GATEWAY_PORT }}':localhost:9090 -i /run/SINK_SSHRELAY_PRIVATE -o ExitOnForwardFailure=yes sshgw@${{ env.RELAY_HOST }} 'sleep infinity'
+
+      - name: Block workflow from finishing
+        run: sleep 4h
+
+      - name: Cleanup ssh process
+        if: ${{ always() }}
```

Comment:
Doesn't hurt, but the `if:` condition does not need the  `${{ }}`, it's always a condition. You didn't use it in the `if:`s above either.